torch/csrc/jit/mobile/import.cpp

```diff
@@ -149,12 +149,8 @@ c10::IValue BytecodeDeserializer::readArchive(const std::string& archive_name) {
   };
 
   auto class_resolver = [&](const c10::QualifiedName& qn) {
-    if (compilation_unit_->get_class(qn) == nullptr) {
```

Comment:
Looking at the APIs closely, I think we should make the following changes:
* class_resolver_ returns TypePtr, not StrongTypePtr
* obj_loader_ takes TypePtr, not StrongTypePtr
* the implementation of obj_loader_ can construct a StrongTypePtr because it can have a reference to the compilation unit. 

This enables us to keep all the standard asserts in StrongTypePtr while still allowing mobile to leave the type null.